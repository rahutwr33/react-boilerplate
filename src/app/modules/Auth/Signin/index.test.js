import { shallow ,mount } from "enzyme";
import SignIn from './index'
import React from 'react'

import { BrowserRouter as  Router} from 'react-router-dom'
describe('Signin component rendering', () => {
    const wrapper = shallow(<Router><SignIn/></Router>);

    it('Render App properly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    it('should have an email field', () => {
        expect(wrapper.find('#email').exists())
    });

    it('should have an password field', () => {
        expect(wrapper.find('#password').length).equals(1)
    });
    
})

