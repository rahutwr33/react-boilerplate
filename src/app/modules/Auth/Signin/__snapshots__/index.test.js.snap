// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin component rendering Render App properly 1`] = `"<main class=\\"MuiContainer-root MuiContainer-maxWidthXs\\" id=\\"signin\\"><div class=\\"makeStyles-paper-1\\"><div class=\\"MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z\\"></path></svg></div><h1 class=\\"MuiTypography-root MuiTypography-h5\\">Sign in</h1><form class=\\"makeStyles-form-3\\" novalidate=\\"\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"email\\" id=\\"email-label\\">Email Address<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" autoComplete=\\"email\\" autofocus=\\"\\" id=\\"email\\" name=\\"email\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-117 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-119\\"><span>Email Address *</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"password\\" id=\\"password-label\\">Password<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"password\\" aria-invalid=\\"false\\" autoComplete=\\"current-password\\" id=\\"password\\" name=\\"password\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-117 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-119\\"><span>Password *</span></legend></fieldset></div></div><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-133 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"checkbox\\" class=\\"PrivateSwitchBase-input-136\\" value=\\"remember\\" data-indeterminate=\\"false\\"/><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path></svg></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">Remember me</span></label><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Sign In</span></button><div class=\\"MuiGrid-root MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary\\" href=\\"/forgotpassword\\">Forgot password?</a></div><div class=\\"MuiGrid-root MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary\\" href=\\"/signup\\">Don&#x27;t have an account? Sign Up</a></div></div></form></div><footer class=\\"makeStyles-footer-287\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-alignCenter\\">React Authentication BoilerPlate</h6><p class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter\\"></p><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter\\">Copyright © <a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit\\" href=\\"https://material-ui.com/\\">Your Website</a> 2020.</p></div></footer></main>"`;
